### ISSUE TITLE
Docs4NIST.yml Rejected on Push

### ISSUE THREAD

**Issue by drfrpjr** [usnistgov/Docs4NIST#22]
Hi,

When I push .github/workflows/Docs4NIST.yml  I get an error message that the push is rejected as shown below. I am not sure about the workflow scope error. 

phelan@P862431 MINGW64 /c/GitHub/Projects/WebFF-Documentation/.github/workflows (master)
$ git push -u origin master
Enumerating objects: 6, done.
Counting objects: 100% (6/6), done.
Delta compression using up to 8 threads
Compressing objects: 100% (3/3), done.
Writing objects: 100% (5/5), 704 bytes | 704.00 KiB/s, done.
Total 5 (delta 1), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (1/1), completed with 1 local object.
To https://github.com/usnistgov/WebFF-Documentation.git
**! [remote rejected] master -> master (refusing to allow a Personal Access Token to create or update workflow `.github/workflows/Docs4NIST
.yml` without `workflow` scope)
error: failed to push some refs to 'https://github.com/usnistgov/WebFF-Documentation.git'**

**Comment by guyer** [comment]
OK, I've got a couple of meetings, but I'll take a look at this later this afternoon

________________________________________
From: Frederick R. Phelan Jr. ***@***.***>
Sent: Friday, February 23, 2024 12:50 PM
To: usnistgov/Docs4NIST
Cc: Subscribed
Subject: [usnistgov/Docs4NIST] Docs4NIST.yml Rejected on Push (Issue #22)

Hi,

When I push .github/workflows/Docs4NIST.yml I get an error message that the push is rejected as shown below. I am not sure about the workflow scope error.

***@***.*** MINGW64 /c/GitHub/Projects/WebFF-Documentation/.github/workflows (master)
$ git push -u origin master
Enumerating objects: 6, done.
Counting objects: 100% (6/6), done.
Delta compression using up to 8 threads
Compressing objects: 100% (3/3), done.
Writing objects: 100% (5/5), 704 bytes | 704.00 KiB/s, done.
Total 5 (delta 1), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (1/1), completed with 1 local object.
To https://github.com/usnistgov/WebFF-Documentation.git
! [remote rejected] master -> master (refusing to allow a Personal Access Token to create or update workflow .github/workflows/Docs4NIST .yml without workflow scope)
error: failed to push some refs to 'https://github.com/usnistgov/WebFF-Documentation.git'

**Comment by drfrpjr** [comment]
It's OK, I solved this particular issue. Thanks.

~Fred
--------------------------------------
Frederick R. Phelan Jr., Ph.D.
Functional Polymers Group
NIST/MML/MSED
Gaithersburg, MD, 20899
--------------------------------------
Sent from Telework WS

**Comment by guyer** [comment]
OK. Anything that needs to be documented better?

**Comment by drfrpjr** [comment]
This was a simple case that the personal access token did not have workflow permission enabled. I edited and updated it and then it worked fine.

I will say that the one thing that is not clear to me from your docs and the related pages docs are whether you need to manually add the header and footer stuff or if it gets inserted automatically, cf., https://github.com/usnistgov/pages-root/wiki/Adding-NIST-required-branding-and-features-to-your-site

I'm running with the idea that the line "include-header-footer: true" in the Docs4NIST YML file handles it although I don't see that specifically stated anywhere. Maybe I missed it?

Anyway, I'm working now on editing the make.bat, conf.py and Makefile to match your stuff and then adding the webhook to give it a shot.

Also, I'm assuming that the themes are the same as the default sphinx themes which makes sense.

I'll get you know of any other Docs4NIST documentation needs when I get through this step. But, those are my meandering thoughts.

Thanks,

~Fred
--------------------------------------
Frederick R. Phelan Jr., Ph.D.
Functional Polymers Group
NIST/MML/MSED
Gaithersburg, MD, 20899
--------------------------------------
Sent from Telework WS

**Comment by drfrpjr** [comment]
Yes, I saw that page but my point was that I feel a clearer documentation of the "include-header-footer: true" action is warranted.

**Comment by guyer** [comment]
!?

It says 

> include-header-footer
>
> Whether to insert the [NIST header and footer](https://pages.nist.gov/nist-header-footer). This content conflicts with, e.g., [sphinx_rtd_theme](https://sphinx-rtd-theme.readthedocs.io/).

What in the world isn't clear about that?

**Comment by drfrpjr** [comment]
I'm new to this, so things aren't quite so clear and certain to me as they are for you who have been looking that this for a good deal longer.

I'm also quite familiar with the text you cut and pasted below.

That text is literally referencing the page: https://github.com/usnistgov/pages-root/wiki/Adding-NIST-required-branding-and-features-to-your-site

So, when I was trying to figure this out, I clicked that link thinking it was relevant.

There, it says in part: "you add a couple of lines to load the script and style sheets to your master page template."

Which seems to imply the user needs to manually add instructions to complete this action.

So, I see between these two things a conflict and lack of clarity.

Do you 1) manually add code as the reference page says and the "true" setting allows it to be actionable; or 2) does setting this "true" take care of all header-footer needs in one fell swoop?

Why not just say "Use this to automatically insert the NIST header and footer without any manual insertions to the doc files."

Or, something to that effect.

This would clarify the information on the branding page.

Or, take out the link to the branding page. This action supersedes that information.

Note, I only said this because you said: "OK. Anything that needs to be documented better?"

To me that is what needs to be documented better.

Thanks,

~Fred
--------------------------------------
Frederick R. Phelan Jr., Ph.D.
Functional Polymers Group
NIST/MML/MSED
Gaithersburg, MD, 20899
--------------------------------------
Sent from Laptop
